packages/react/src/AvatarStack/AvatarStack.tsx

```diff
@@ -307,16 +334,44 @@ const AvatarStack = ({
     sxProp as SxProp,
   )
 
-  return (
-    <AvatarStackWrapper count={count} className={wrapperClassNames} sx={avatarStackSx}>
+  const AvatarStackBody = ({children}: React.ComponentPropsWithoutRef<'div'>) => {
+    if (enabled) {
+      return (
+        <div
+          data-disable-expand={disableExpand ? '' : undefined}
+          className={clsx(bodyClassNames, classes.AvatarStackBody)}
+          tabIndex={!hasInteractiveChildren && !disableExpand ? 0 : undefined}
+          ref={stackContainer}
+        >
+          {' '}
+          {children}
+        </div>
+      )
+    }
+    return (
       <Box
         className={bodyClassNames}
         tabIndex={!hasInteractiveChildren && !disableExpand ? 0 : undefined}
         ref={stackContainer}
       >
         {' '}
-        {transformChildren(children)}
+        {children}
       </Box>
+    )
+  }
+
+  return (
+    <AvatarStackWrapper
+      count={enabled ? undefined : count}
+      data-avatar-count={enabled ? (count > 3 ? '3+' : count) : undefined}
+      data-align-right={enabled && alignRight ? '' : undefined}
+      data-responsive={enabled && (!size || isResponsiveValue(size)) ? '' : undefined}
+      className={clsx(wrapperClassNames, {[classes.AvatarStack]: enabled})}
+      // @ts-ignore - it's not allowing CSS properties here
+      style={enabled ? (getResponsiveAvatarSizeStyles() as React.CSSProperties) : undefined}
```

Comment:
Would it help if `getResponseiveAvatarSizeStyles` returned `React.CSSProperties` or not really? Wasn't sure what would help out best with the type casting in the file, either way works.